src/config.rs

```diff
@@ -61,8 +61,96 @@ struct Root {
     clouds: Clouds,
 }
 
-fn find_config() -> Option<PathBuf> {
-    let current = Path::new("./clouds.yaml");
+/// Merge two nested serde_yaml::Mapping structs
+/// The values from src are merged into dest. Values in src override values in dest.
+fn merge_mappings(src: &serde_yaml::Mapping, dest: &mut serde_yaml::Mapping) {
+    for (src_key, src_value) in src.iter() {
+        if src_value.is_mapping()
+            && dest.contains_key(src_key)
+            && dest.get(src_key).unwrap().is_mapping()
+        {
+            merge_mappings(
+                src_value.as_mapping().unwrap(),
+                dest.get_mut(src_key).unwrap().as_mapping_mut().unwrap(),
+            );
+        } else {
+            match dest.insert(src_key.to_owned(), src_value.to_owned()) {
+                // Ignore the Option returned
+                _ => {}
+            }
+        }
+    }
+}
+
+// Inject profiles from clouds-public.yaml into clouds.yaml and return it in a new Value
+fn inject_profiles(
+    clouds_public: &serde_yaml::Mapping,
+    clouds: &serde_yaml::Mapping,
+) -> Result<serde_yaml::Value, Error> {
+    let mut temp_mapping = serde_yaml::Mapping::new();
+    match temp_mapping.insert(
+        "clouds".into(),
+        serde_yaml::Value::Mapping(serde_yaml::Mapping::new()),
+    ) {
+        _ => {} // ignore returned Option
+    };
+
+    let clouds_mapping = clouds.get(&"clouds".into());
+    let clouds_public_mapping = clouds_public.get(&"public-clouds".into());
+
+    if clouds_mapping.is_some()
+        && clouds_mapping.unwrap().is_mapping()
+        && clouds_public_mapping.is_some()
+        && clouds_public_mapping.unwrap().is_mapping()
```

Comment:
could you maybe change it to

```rust
if let Some(mapping) = cloud_mapping {
    if let Some(public) = cloud_public_mapping {
...
```
? This way you'll avoid so many `unwrap`'s (which are potentially dangerous if done wrong).